### ISSUE TITLE
Effect.if evaluates onTrue when condition is false

### ISSUE THREAD

**Issue by alex-dixon** [Effect-TS/effect#2345]
### What version of Effect is running?

2.2.3

### What steps can reproduce the bug?

```typescript
import { Effect } from 'effect'

const f = (s: string) => Effect.succeed(s)

const foo: { bar?: { baz: string } } = {}

Effect.runSync(
  Effect.if('bar' in foo, {
    onTrue: f(foo.bar!.baz),
    onFalse: Effect.fail('oops'),
  })
)
```

### What is the expected behavior?

onTrue does not evaluate when truth test is false

### What do you see instead?

```
/repro/scratch.ts:9
    onTrue: f(foo.bar!.baz),
                       ^
TypeError: Cannot read properties of undefined (reading 'baz')
    at Object.<anonymous> (/repro/scratch.ts:9:24)
    at Module._compile (node:internal/modules/cjs/loader:1368:14)
    at Module.m._compile (/repro/node_modules/ts-node/src/index.ts:1618:23)
```

**Comment by tim-smart** [comment]
You will need to wrap it with Effect.suspend

**Comment by mikearnaldi** [comment]
> You will need to wrap it with Effect.suspend

might be worth considering if conditional branches should be lazy, while the fact that the branch would evaluate is pretty obvious from JS eager evaluation semantics the golden path should be not to evaluate the branch

**Comment by tim-smart** [comment]
Closed by #2451
